src/sonymn_int.cpp

```diff
@@ -791,21 +792,87 @@ namespace Exiv2 {
-        {  0x16, "FocusMode"         , N_("Focus Mode")         , N_("Focus Mode")         , sony2FpId, makerTags, unsignedByte, 1, printValue},
-        {  0x17, "AFAreaMode"        , N_("AF Area Mode")       , N_("AF Area Mode")       , sony2FpId, makerTags, unsignedByte, 1, printValue},
-        {  0x2d, "FocusPosition2"    , N_("Focus Position 2")   , N_("Focus Position 2")   , sony2FpId, makerTags, unsignedByte, 1, printValue},
+        {  0x04, "AmbientTemperature", N_("Ambient temperature"), N_("Temperature of the surroundings (in degrees Celsius)"), sony2FpId, makerTags, signedByte, 1, printTemperatureInDegC},
+        {  0x16, "FocusMode"         , N_("Focus mode")         , N_("Focus mode")          , sony2FpId, makerTags, unsignedByte, 1, printSony2FpFocusMode},
+        {  0x17, "AFAreaMode"        , N_("AF area mode")       , N_("Auto focus area mode"), sony2FpId, makerTags, unsignedByte, 1, EXV_PRINT_TAG(sony2FpAFAreaMode)},
+        {  0x2d, "FocusPosition2"    , N_("Focus position 2")   , N_("Focus position 2")    , sony2FpId, makerTags, unsignedByte, 1, printSony2FpFocusPosition2},
         // End of list marker
-        {0xffff, "(Unknownsony2FpTag)", "(Unknownsony2FpTag)"   , "(Unknownsony2FpTag)"    , sony2FpId, makerTags, unsignedByte, 1, printValue},
+        {0xffff, "(UnknownSony2FpTag)", "(UnknownSony2FpTag)"   , "(UnknownSony2FpTag)"     , sony2FpId, makerTags, unsignedByte, 1, printValue},
     };
 
     const TagInfo* SonyMakerNote::tagListFp()
     {
         return tagInfoFp_;
     }
 
+    std::ostream& SonyMakerNote::printSony2FpFocusMode(std::ostream& os, const Value& value, const ExifData*)
+    {
+        if (value.count() != 1)
+            os << value;
+        else {
+            long val = (value.toLong() & 0x7F);
+            if (val == 0)
+                os << N_("Manual");
+            else if (val == 2)
+                os << N_("AF-S");
+            else if (val == 3)
+                os << N_("AF-C");
+            else if (val == 4)
+                os << N_("AF-A");
+            else if (val == 6)
+                os << N_("DMF");
+            else
+                os << "(" << val << ")";
+        }
+
+        return os;
+    }
+
+    std::ostream& SonyMakerNote::printSony2FpFocusPosition2(std::ostream& os, const Value& value, const ExifData* metadata)
+    {
+        if (value.count() != 1)
+            os << "(" << value << ")";
+        else {
+            auto pos = metadata->findKey(ExifKey("Exif.Image.Model"));
+            if (pos == metadata->end())
+                return os << "(" << value << ")";
+
+            // Ranges of models that do not support this tag
+            const char* models[] = { "DSC-", "Stellar" };
+
+            std::string model = pos->toString();
+            for (auto&& m : models) {
+                if (model.find(m) == 0) {
```

Comment:
Please could you use `startsWith` here too? Sorry for highlighting it last time.

Reply:
I have added this.